### ISSUE TITLE
Deserializing Either doesn't work when both types are POCOs

### ISSUE THREAD

**Issue by brcrista** [recorefx/RecoreFX#114]
Consider

```cs
class Person
{
    public string Name { get; set; }
    public int Age { get; set; }
}

class Address
{
    public string Street { get; set; }
    public string Zip { get; set; }
}

JsonSerializer.Deserialize<Either<Person, Address>>("{\"Street\":\"123 Main St\",\"Zip\":\"12345\"}")
```

This will always try to deserialize the JSON as a `Person`, leaving all the fields as default values.

A workaround is to add an "undiscriminated union" with some custom logic to decide which type to create based on its values:

```cs
class PersonAddress
{
    public string Name { get; set; }
    public int Age { get; set; }
    public string Street { get; set; }
    public string Zip { get; set; }

    public Either<Person, Address> ToEither()
    {
        if (Name != null)
        {
            return new Person
            {
                Name = Name,
                Age = Age
            };
        }
        else
        {
            return new Address
            {
                Street = Street,
                Zip = Zip
            };
        }
    }
}
```

**Comment by brcrista** [comment]
If there's no way to handle this gracefully, it might be best just to throw an exception from the `EitherConverter` constructor to avoid bugs.

**Comment by brcrista** [comment]
Fixed by #117
